src/core/lib/resource_quota/memory_quota.h

```diff
@@ -327,9 +355,14 @@ class BasicMemoryQuota final
   std::atomic<intptr_t> free_bytes_{kInitialSize};
   // The total number of bytes in this quota.
   std::atomic<size_t> quota_size_{kInitialSize};
+  // Assign a unique ID to each allocator.
+  std::atomic<uint64_t> curr_allocator_id_{0};
 
   // Reclaimer queues.
   ReclaimerQueue reclaimers_[kNumReclamationPasses];
+  // List of all allocators sorted into 2 buckets, small (<100 KB free bytes)
+  // and large (>500 KB free bytes).
+  std::array<AllocatorBucket, 2> allocators_;
```

Comment:
I don't think we're getting benefit from an array here, let's just go with `AllocatorBucket big_allocator_; AllocatorBucket small_allocator_;`?